### PR TITLE
Rest-server all requests and reports endpoint

### LINKED ISSUE #559: Rest-server all requests and reports endpoint
- [ ] Querier types/keeper to get all requests and reports in a range of given ID.
- [ ] A REST endpoint to get the requests and reports information list given the range of ID.

### PR DESCRIPTION
**Pull request by taobun**
fixed #559

### CODE REVIEW

File: chain/x/zoracle/client/rest/query.go [hunk 1]
```diff
@@ -39,64 +40,74 @@ func buildRequestRESTInfo(
...
+				if ok {
+					txReportMap[string(msgReport.Sender)] = buildTxDetail(&report)
+					break
+				}
 			}
 		}
 	}
 
-	return request, nil
+	request.Reports = make([]ReportDetail, 0)
```
Comment (sorawit) [inline chain/x/zoracle/client/rest/query.go:90]: I think this should be nil, not empty list if withReportTx is false.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

File: chain/x/zoracle/internal/keeper/querier.go [hunk 2]
```diff
@@ -78,16 +74,73 @@ func queryRequest(ctx sdk.Context, path []string, req abci.RequestQuery, keeper
...
+	id, err := strconv.ParseInt(path[0], 10, 64)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("wrong format for requestid %s", err.Error()))
+	}
+
+	request, sdkErr := buildRequestQuerierInfo(ctx, keeper, id)
+	if sdkErr != nil {
+		return nil, sdkErr
+	}
+	res, err := codec.MarshalJSONIndent(keeper.cdc, request)
```
Comment (sorawit) [inline chain/x/zoracle/internal/keeper/querier.go]: Yeah same here. Let's use MustMarshalJSON?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/rest/query.go [hunk 3]
```diff
@@ -39,64 +40,74 @@ func buildRequestRESTInfo(
...
+	}
+
+	return request, nil
 }
 
-func getRequestHandler(cliCtx context.CLIContext, storeName string) http.HandlerFunc {
+func getRequestByIDHandler(cliCtx context.CLIContext, storeName string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
 		reqID := vars[requestID]
```
Comment (sorawit) [inline chain/x/zoracle/client/rest/query.go]: See. Confusing!

File: chain/x/zoracle/client/rest/rest.go [hunk 4]
```diff
@@ -14,7 +14,8 @@ const (
 )
 
 func RegisterRoutes(cliCtx context.CLIContext, r *mux.Router, storeName string) {
-	r.HandleFunc(fmt.Sprintf("/%s/request/{%s}", storeName, requestID), getRequestHandler(cliCtx, storeName)).Methods("GET")
+	r.HandleFunc(fmt.Sprintf("/%s/request/{%s}", storeName, requestID), getRequestByIDHandler(cliCtx, storeName)).Methods("GET")
```
Comment (sorawit) [inline chain/x/zoracle/client/rest/rest.go]: Can you rename variable to `requestIDTag`? Having variable name `requestID` on top scope is confusing.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]

File: chain/x/zoracle/internal/types/querier.go [hunk 5]
```diff
@@ -27,19 +28,22 @@ func (u64a U64Array) String() string {
 }
 
 type RequestQuerierInfo struct {
+	RequestID       int64                          `json:id`
```
Comment (evilpeach) [inline chain/x/zoracle/internal/types/querier.go]: It should be `json:"id"`

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**